src/bg_daemon/fetchers/imgurfetcher.py

```diff
@@ -226,27 +227,69 @@ def _select_image(self, galleries):
                 except IndexError:
                     return None
 
+            # if the "image" is actually an album, get the topmost image from
+            # the album
+            if isinstance(selected_image, GalleryAlbum):
+                selected_image = self._get_image_from_album(selected_image)
+                if selected_image is None:
+                    continue
+
+            logger.debug("Selecting Image {}".format(selected_image.title))
             attempts += 1
             if attempts > 30:
                 return None
 
             if selected_image.width < self.min_width:
+                logger.debug("Rejecting due to width...")
                 continue
 
             if selected_image.height < self.min_height:
+                logger.debug("Rejecting due to height...")
                 continue
 
-            for word in self.blacklist_words:
-                if word in selected_image.title:
-                    continue
+            bad_title = False
```

Comment:
Instead of using `bad_title` and repeating the `bad_title = True` below, you could use a [`for`/`else`](https://docs.python.org/2/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops) statement.